quinn-proto/src/crypto.rs

```diff
@@ -78,6 +77,24 @@ impl TlsSession {
             },
         }
     }
+
+    pub fn write_handshake(&mut self, buf: &mut Vec<u8>) -> Result<Option<Crypto>, TLSError> {
```

Comment:
This error case seems to neither be generated nor handled. I guess the idea is to future proof the type signature in case we ever end up with a handshake that doesn't validate incoming data up front?

Reply:
Oh, good catch. Removed that and simplified the code some more in the process.